### DISCUSSION TITLE
linux binary systemctl service

### DISCUSSION THREAD

**Discussion by 91320113** [qdm12/ddns-updater#693] [Q&A] [answered]
Hello,

I'm trying to set up a service for ddns-updater to make sure it's always up and running.

```ini
[Unit]
Description=DDNS Updater
After=network.target network-online.target
Requires=network-online.target
StartLimitBurst=5
StartLimitIntervalSec=10

[Service]
Type=notify
Restart=on-failure
RestartSec=1
ExecStart=/usr/bin/ddns-updater/ddns-updater --period=20s --update-cooldown-period=1m
#ExecReload=
#TimeoutStopSec=5s
#ProtectSystem=full
AmbientCapabilities=CAP_NET_ADMIN CAP_NET_BIND_SERVICE

[Install]
WantedBy=multi-user.target
```

But when starting the service it's stuck on status activating, please advice.

Thanks

**Comment by qdm12** [comment]
Perhaps change the type to simple?

**Reply by 91320113** [reply]
Same issue, seems that the service not getting exit 1 code and timeout.

I made sh file:

#!/bin/bash

cd /usr/bin/ddns-updater && ./ddns-updater --period=1m --update-cooldown-period=2m

exit 1

and pointed the service to it now it's working but not sure if the service will catch the failure if happens, I tried to kill the process and nothing happened.

Would be great if this could be implemented to make sure the service always on and run some hooks if not. 

Thank you for your immediate response.

**Comment by 91320113** [comment, marked as answer]
/etc/systemd/system/ddns-updater.service:

```
[Unit]
Description=DDNS Updater
After=network.target network-online.target
Requires=network-online.target
StartLimitBurst=5
StartLimitIntervalSec=10

[Service]
Type=simple
Restart=on-failure
RestartSec=1
ExecStart= /usr/bin/ddns-updater/init.sh
AmbientCapabilities=CAP_NET_ADMIN CAP_NET_BIND_SERVICE

[Install]
WantedBy=multi-user.target
```


/usr/bin/ddns-updater/init.sh:
```
#!/bin/bash
cd /usr/bin/ddns-updater && ./ddns-updater --period=20s --update-cooldown-period=1m
exit 1
```

This works and restarts if kill -9 the process, ugly solution but working for now.

Thank you

**Reply by qdm12** [reply]
Does it work with type notify as well?

By the way, if you don't update the binary program and don't change your settings, there is near zero reason it would crash or exit really 😉 It would log warnings or errors if it fails to update a domain, but it wouldn't exit.
